src/dotnet/APIView/APIView/NamedTypeAPIV.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Microsoft.CodeAnalysis;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+
+namespace APIView
+{
+    public class NamedTypeAPIV
+    {
+        public string Name { get; }
+        public string Type { get; }
+        public string EnumUnderlyingType { get; }
+
+        public ImmutableArray<EventAPIV> Events { get; }
+        public ImmutableArray<FieldAPIV> Fields { get; }
+        public ImmutableArray<string> Implementations { get; }
+        public ImmutableArray<MethodAPIV> Methods { get; }
+        public ImmutableArray<NamedTypeAPIV> NamedTypes { get; }
+        public ImmutableArray<PropertyAPIV> Properties { get; }
+
+        /// <summary>
+        /// Construct a new namedType instance, represented by the provided symbol.
+        /// </summary>
+        /// <param name="symbol">The symbol representing the named type.</param>
+        public NamedTypeAPIV(INamedTypeSymbol symbol)
+        {
+            this.Name = symbol.Name;
+            this.Type = symbol.TypeKind.ToString().ToLower();
```

Comment:
I would store this as an enum, it's easier to work with and is less error-prone.